### PR TITLE
chore: add more explicit types

### PR DESCRIPTION
**Pull request by dsherret**
This will further help with deno doc output.

### CODE REVIEW

File: yaml/schema/core.ts [hunk 1]
```diff
@@ -9,6 +9,6 @@ import { json } from "./json.ts";
 
 // Standard YAML's Core schema.
 // http://www.yaml.org/spec/1.2/spec.html#id2804923
-export const core = new Schema({
+export const core: Schema = new Schema({
```
Comment (dsherret) [inline yaml/schema/core.ts:12]: Can't be easily inferred because `Schema` could have a type parameter.

File: log/mod.ts [hunk 2]
```diff
@@ -431,15 +431,23 @@ const state = {
  * For examples check source code of {@linkcode FileHandler}`
  * and {@linkcode TestHandler}.
  */
-export const handlers = {
+export const handlers: {
+  BaseHandler: typeof BaseHandler;
+  ConsoleHandler: typeof ConsoleHandler;
+  WriterHandler: typeof WriterHandler;
+  FileHandler: typeof FileHandler;
+  RotatingFileHandler: typeof RotatingFileHandler;
```
Comment (dsherret) [inline log/mod.ts:439]: Slightly unfortunate, but this is not too bad.

File: bytes/last_index_of_needle.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@
 export function lastIndexOfNeedle(
   source: Uint8Array,
   needle: Uint8Array,
-  start = source.length - 1,
+  start: number = source.length - 1,
```
Comment (iuioiua) [inline bytes/last_index_of_needle.ts:23]: Is this required? `deno doc --lint bytes/last_index_of_needle.ts` reports no errors before this change.

File: io/slice_long_to_bytes.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@
  */
 export function sliceLongToBytes(
   d: number,
-  dest = Array.from<number>({ length: 8 }),
+  dest: number[] = Array.from<number>({ length: 8 }),
```
Comment (iuioiua) [inline io/slice_long_to_bytes.ts:13]: Ditto. Is this needed?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Explicit types for arguments with default values don't appear to be needed.

File: bytes/last_index_of_needle.ts (hunk 3 above)
Comment (dsherret) [reply bytes/last_index_of_needle.ts:23]: I think there's a bug in `deno doc --lint` related to this. I'm running a separate analyzer (it is an optimization to type checking).

See here how the type is missing: https://deno.land/std@0.210.0/bytes/last_index_of_needle.ts?s=lastIndexOfNeedle

File: io/slice_long_to_bytes.ts (hunk 4 above)
Comment (dsherret) [reply io/slice_long_to_bytes.ts:13]: See here how `dest` is unsupported. It's a bug in `deno doc --lint`:

https://deno.land/std@0.210.0/io/slice_long_to_bytes.ts?s=sliceLongToBytes

File: bytes/last_index_of_needle.ts (hunk 3 above)
Comment (iuioiua) [reply bytes/last_index_of_needle.ts:23]: Oh, I see. Perhaps we should fix that bug instead of accounting for it. WDYT?

Comment (dsherret) [reply bytes/last_index_of_needle.ts:23]: I opened https://github.com/denoland/deno_doc/issues/435

Doing a big refactor of everything at the moment (which is what helped spot this), so I'll add a test and investigate in there later.

Comment (iuioiua) [reply bytes/last_index_of_needle.ts:23]: Perhaps let's continue to include these explicit types on arguments with default values. It'll fix the docs sooner. We can then come behind later on and clean these up. I can create an issue once this PR is merged.

File: log/mod.ts [hunk 5]
```diff
@@ -431,15 +431,23 @@ const state = {
  * For examples check source code of {@linkcode FileHandler}`
  * and {@linkcode TestHandler}.
  */
-export const handlers = {
+export const handlers: {
```
Comment (dsherret) [inline log/mod.ts:434]: Actually, going to revert this. Deno doc understands it so I should be able to get the other thing to as well.

File: bytes/last_index_of_needle.ts (hunk 3 above)
Comment (dsherret) [reply bytes/last_index_of_needle.ts:23]: I think this is all the public API that's affected. Everything else seems to be good. After this PR I have Deno type checking a subset of the code properly in the new registry.

**Comment by iuioiua** [comment]
Towards #3764 and #3878.

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**